packages/core/cache/src/Cache.js

```diff
@@ -105,4 +76,35 @@ export class Cache {
   }
 }
 
-export default new Cache();
+// Cache for whether a cache dir exists
+const existsCache: Set<FilePath> = new Set();
+const cacheByDir: DefaultMap<FilePath, Cache> = new DefaultMap(dir => {
+  return new Cache(dir);
```

Comment:
this `DefaultMap` has the potential to grow unbounded. I considered exposing a way of freeing references to Caches, but this could be problematic with multiple concurrent parcels working with the same cache, as they would share the same `invalidations` state. Maybe we could count references, and free the cache when users dispose of them?

On the other hand, these objects aren't large and we're unlikely to have a large number of parcels working on large numbers of cache dirs.

Reply:
I don't think we need to keep the invalidations state in Cache. That is left over from when the Cache was more coupled to how Parcel transformations were cached.